fgpyo/sam/builder.py

```diff
@@ -405,15 +420,49 @@ def add_pair(
 
         name = name if name is not None else self._next_name()
 
+        # Valid parameterizations for contig mapping (backward compatible):
+        # - chrom, start1, start2
+        # - chrom, start1
+        # - chrom, start2
+        # Valid parameterizations for contig mapping (new):
+        # - chrom1, start1, chrom2, start2
+        # - chrom1, start1
+        # - chrom2, start2
+
+        # Fail if chrom is specified with chrom1 or chrom2
+        if chrom != sam.NO_REF_NAME and (chrom1 is not None or chrom2 is not None):
+            raise ValueError("chrom cannot be specified together with chrom1 or chrom2.")
+
+        # Backwards compatibility
+        # Use `chrom` for both reads if neither `chrom1` nor `chrom2` are specified.
```

Comment:
I _think_ this is overly restrictive, but I'm having a hard time thinking through all the cases.  Would it not suffice to do something like:

```
if chrom is None and (chrom1 is not None or chrom2 is not None):
    raise ValueError("Cannot use chrom in combination with chrom1 or chrom2")

if chrom is None and chrom1 is None and chrom2 is None:
    chrom = sam.NO_REF_NAME

chrom1 = next(c for c in (chrom2, chrom) if c is not None)
chrom2 = next(c for c in (chrom1, chrom) if c is not None)
```

Reply:
Yeah I find this clearer.

Just pushed, with additional checks to ensure a chromosome is always specified when providing a start position

https://github.com/fulcrumgenomics/fgpyo/pull/61/commits/8a55ab18d68aa57b09ec59edbada88e8ef30226f